1_httpshim/src/httpshim.ts

```diff
@@ -135,16 +135,22 @@ export class HttpClient {
             req.setHeader('x-lat', geo.Lat);
             req.setHeader('x-long', geo.Long);
         }
-        req.setHeader('content-type', 'application/json');
         req.setHeader('accept', 'application/json');
         if (authHeader != null) {
             req.setHeader('Authorization', authHeader);
         }
 
         if (body != null) {
-            var dataJson: string = JSON.stringify(body);
-            req.end(dataJson, 'utf8');
+            if (typeof body === 'object') {
+                req.setHeader('content-type', 'application/json');
+                var dataJson: string = JSON.stringify(body);
+                req.end(dataJson, 'utf8');
+            } else {
+                req.setHeader('content-type', 'application/octet-stream');
```

Comment:
What caller type causes us to go down this path?

Reply:
Any call that has a body, and that body is not a JS object.

Anything with a valid JS object body, or no body at all, won't reach that part.